src/Appwrite/Platform/Workers/Builds.php

```diff
@@ -207,7 +207,66 @@ protected function buildDeployment(Device $deviceForFunctions, Func $queueForFun
         }
 
         try {
-            if ($isNewBuild && $isVcsEnabled) {
+            if($isNewBuild && !$isVcsEnabled) {
+                // Non-vcs+Template
+
+                $templateRepositoryName = $template->getAttribute('repositoryName', '');
+                $templateOwnerName = $template->getAttribute('ownerName', '');
+                $tempalteVersion = $template->getAttribute('version', '');
+
+                $templateRootDirectory = $template->getAttribute('rootDirectory', '');
+                $templateRootDirectory = \rtrim($templateRootDirectory, '/');
+                $templateRootDirectory = \ltrim($templateRootDirectory, '.');
+                $templateRootDirectory = \ltrim($templateRootDirectory, '/');
+
+                if (!empty($templateRepositoryName) && !empty($templateOwnerName) && !empty($tempalteVersion)) {
+                    $stdout = '';
+                    $stderr = '';
+
+                    // Clone template repo
+                    $tmpTemplateDirectory = '/tmp/builds/' . \escapeshellcmd($buildId) . '-template';
+                    $gitCloneCommandForTemplate = $github->generateCloneCommand($templateOwnerName, $templateRepositoryName, $tempalteVersion, GitHub::CLONE_TYPE_TAG, $tmpTemplateDirectory, $templateRootDirectory);
+                    $exit = Console::execute($gitCloneCommandForTemplate, '', $stdout, $stderr);
+
+                    if ($exit !== 0) {
+                        throw new \Exception('Unable to clone code repository: ' . $stderr);
+                    }
+
+                    // Ensure directories
+                    Console::execute('mkdir -p ' . $tmpTemplateDirectory . '/' . $templateRootDirectory, '', $stdout, $stderr);
+
+                    $tmpPathFile = $tmpTemplateDirectory . '/code.tar.gz';
+
+                    $localDevice = new Local();
+
+                    if (substr($tmpTemplateDirectory, -1) !== '/') {
+                        $tmpTemplateDirectory .= '/';
+                    }
+
+                    $directorySize = $localDevice->getDirectorySize($tmpTemplateDirectory);
+                    $functionsSizeLimit = (int)System::getEnv('_APP_FUNCTIONS_SIZE_LIMIT', '30000000');
+                    if ($directorySize > $functionsSizeLimit) {
+                        throw new \Exception('Repository directory size should be less than ' . number_format($functionsSizeLimit / 1048576, 2) . ' MBs.');
```

Comment:
1048576 what exactly is this number?

Reply:
1024 * 1024. Not a new code, copy&paste from different place.